subsys/shell/shell_uart.c

```diff
@@ -78,10 +82,13 @@ static void uart_rx_handle(struct device *dev,
 
 			rd_len = uart_fifo_read(dev, &dummy, 1);
 #ifdef CONFIG_MCUMGR_SMP_SHELL
-			/* Divert this byte from shell handling if it
-			 * is part of an mcumgr frame.
+			/* If successful in getting byte from the fifo, try
+			 * feeding it to SMP as a part of mcumgr frame.
 			 */
-			smp_shell_rx_byte(&sh_uart->ctrl_blk->smp, dummy);
+			if (rd_len != 0 &&
```

Comment:
I know this is not compulsory but would you mind putting this condition within parentheses?